src/IO/H5/Dat.cpp

```diff
@@ -102,7 +102,25 @@ Dat::Dat(const bool exists, detail::OpenGroup&& group, const hid_t location,
   }
 }
 
-Dat::~Dat() { CHECK_H5(H5Dclose(dataset_id_), "Failed to close dataset"); }
+Dat::~Dat() {
+// The pragmas here are used to suppress the warning that the compile time
+// branch of `ERROR` will always call `terminate()` because it throws an error.
+// Throwing that error is a code path that will never actually be entered at
+// runtime, so we suppress the warning here.
+#pragma GCC diagnostic push
+// Suppresses -Wpragmas warning for gcc builds that do not recognize
+// -Wunknown-warning-option
+#pragma GCC diagnostic ignored "-Wpragmas"
+// Suppresses -Wunknown-warning-option for clang builds that do not recognize
+// -Wterminate
+#pragma GCC diagnostic ignored "-Wunknown-warning-option"
+// Suppresses -Wterminate for gcc-9 and gcc-10 builds
+#pragma GCC diagnostic ignored "-Wterminate"
+// Suppresses -Werror=terminate for gcc-11 builds
+#pragma GCC diagnostic ignored "-Wno-error=terminate"
```

Comment:
Is this needed?  This isn't necessary for any of our other suppressions.  And if it is needed shouldn't it be `#pragma GCC diagnostic ignored "-Werror=terminate"` (no `no-`)?

Reply:
When I previously tried the version you provided, it didn't work and only worked after I updated it to this current version. I can definitely double check myself though

Reply:
And yes it was necessary to suppress a gcc-11 build warning, specifically